Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,78 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
+           string result= "";
+            //condition if only one exactPostTime is given
+            if(exactPostTime.Length == 1)
+            {
+                result = exactPostTime[0];
+                return result;
+            }
+            //
+            for(int i = 0; i < exactPostTime.Length; i++)
+            {
+                for(int j=0; j<exactPostTime.Length; j++)
+                {
+                    if((exactPostTime[i] == exactPostTime[j]) && (showPostTime[i] != showPostTime[j]))
+                    {
+                        return "impossible";
+                    }
+
+                }
+                var exactHour = exactPostTime[i].Split(":")[0];
+                var HH = int.Parse(exactHour);
+                var exactMinute = exactPostTime[i].Split(":")[1];         
+                var MM = int.Parse(exactMinute);
+                var exactSecond = exactPostTime[i].Split(":")[2];
+                var SS = int.Parse(exactSecond);
+                // if showPostTime contain hour then the total number of hour ago is added to HH which shows current hour 
+                if(showPostTime[i].Contains("hours"))
+                {
+                    var shownHour = showPostTime[i].Split(" ")[0];
+                    HH += int.Parse(shownHour);
+                    if(HH > 24)
+                    {
+                        HH -= 24;
+                    }
+                }
+                // if showPostTime contain minutes then the total number of minutes ago is added to MM which shows current minute 
+                if(showPostTime[i].Contains("minutes"))

```

Comment:
Your approach is sound, trying to build a time component from a input
like hour from "hours ago"
min from "mins ago"

But there could be have been multiple "hours ago" or "mins ago" and for each element you are finding "result" and overriding it with the next same "hours ago" or "mins ago", so at the termination of your program you'll not have a lexicographically smaller result.

**Your approach need to be relative in inputs**

And also, consider the scenerio, when input only contains multiple "a few seconds ago".
That is, {"11:11:13","11:11:19","11:11:43"} & {"few seconds ago","few seconds ago".few seconds ago"}

One more suggestion, you could have also used [DateTime](https://docs.microsoft.com/en-us/dotnet/api/system.datetime?view=netframework-4.8)  functions defined in System namespace, they provide you a hell lot of functions which work on date and time.